### PR TITLE
Change some leftover references from external data store to data connection

### PR DESCRIPTION
**Pull request by frant-hartm**
_Data connection_ feature was called _external data store_ in its beta phase (in Hazelcast 5.3). Apparently, there are still some references to the old name left. 

I have tried to capture everything, apart from the names of the pages / links - I don't understand how the redirects work exactly and didn't want to break anything.

### CODE REVIEW

File: docs/modules/integrate/pages/jdbc-connector.adoc [hunk 1]
```diff
@@ -45,11 +45,11 @@ p.readFrom(Sources.jdbc(
 )).writeTo(Sinks.logger());
 ```
 
-You can also use a configured xref:external-data-stores:external-data-stores.adoc#defining-external-data-stores[external data store] as a JDBC Source:
+You can also use a configured xref:external-data-stores:external-data-stores.adoc#defining-external-data-stores[data connection] as a JDBC Source:
```
Comment (oliverhowell) [inline docs/modules/integrate/pages/jdbc-connector.adoc:48]: This page doesn't exist...looks like data-connections-configuration.adoc has the alias and the link should still work

File: docs/modules/sql/pages/mapping-to-jdbc.adoc [hunk 2]
```diff
@@ -64,32 +64,31 @@ In the MySQL database, create a `people` table.
 CREATE TABLE people (id INT PRIMARY KEY, name VARCHAR(100))
 ```
 
-Use the `CREATE MAPPING` statement to tell the JDBC connector the name of your data store configuration.
+On all your members define a data connection of type `JDBC`:
```
Comment (oliverhowell) [inline docs/modules/sql/pages/mapping-to-jdbc.adoc:67]:
```suggestion
Define a data connection of type `JDBC` on all members:
```

**Review by oliverhowell — APPROVED** [review]
Minor comments but approving in advance

File: docs/modules/integrate/pages/jdbc-connector.adoc (hunk 1 above)
Comment (Rob-Hazelcast) [reply docs/modules/integrate/pages/jdbc-connector.adoc]: The link does still work, but the updated link text wasn't showing up in the build for some reason. I've just updated all these xrefs (and any child section refs) to point to the new file.

**Review by amandalindsay — APPROVED** [review]

**Pull request merged**